File: packages/ui-swiftui/Sources/KinetixUI/ScrollArea.swift
//
// ScrollArea.swift — KinetixScrollArea.
//
// Mirrors packages/ui/src/components/scroll-area.tsx. Wraps SwiftUI's
// native `ScrollView` — the custom Radix scrollbar isn't reproduced (the
// system indicators take its place), the same "reuse the platform
// machinery" call the Compose port made.
//

import SwiftUI

public struct KinetixScrollArea<Content: View>: View {
    private let axes: Axis.Set
    private let showsIndicators: Bool
    private let content: Content

    public init(
        _ axes: Axis.Set = .vertical,
        showsIndicators: Bool = true,
        @ViewBuilder content: () -> Content
    ) {
        self.axes = axes
        self.showsIndicators = showsIndicators
        self.content = content()
    }

    public var body: some View {
        ScrollView(axes, showsIndicators: showsIndicators) { content }
    }
}

File: packages/ui-flutter/lib/src/scroll_area.dart
import 'package:flutter/widgets.dart';

/// Mirrors `packages/ui/src/components/scroll-area.tsx`. Wraps
/// `SingleChildScrollView` — the custom Radix scrollbar isn't reproduced
/// (the platform's own indicators take its place), same as the other
/// ports.
class KinetixScrollArea extends StatelessWidget {
  const KinetixScrollArea({super.key, required this.child, this.axis = Axis.vertical});

  final Widget child;
  final Axis axis;

  @override
  Widget build(BuildContext context) {
    return SingleChildScrollView(scrollDirection: axis, child: child);
  }
}
